### ISSUE TITLE
simplify fails to create a reasonable polygon in some cases

### ISSUE THREAD

**Issue by trygveaa** [Turfjs/turf#1845]
When simplifying some polygons, `simplify` creates just a few points around the starting point of the polygon with a shape and size that doesn't resemble the original polygon. I tried simplifying directly with `simplify-js` and the issue doesn't occur with that. I'm not sure what kind of polygons triggers the issue.

Here are the polygons: https://gist.github.com/trygveaa/8613e83e77bf69fefe4a81e5a43e40ca

`original-polygon.geojson` is the original polygon I started with.

`turf-simplify-polygon.geojson` was created with this:

```
turf.simplify(polygon, { tolerance: 0.001, highQuality: true })
```

`simplify-js-polygon.geojson` was created with this:

```
coords = polygon.coordinates[0].map(coord => ({
  x: coord[0],
  y: coord[1]
}))
polygon.coordinates[0] = simplify(coords, 0.001, true).map(coord => [
  coord.x,
  coord.y
])
```

**Comment by GaborFarkas** [comment]
Simplify's cleanLine iterates through the original polygon, and does not consider the last added "clean" point as the previous point. In cases when successive point triplets are close enough to be considered as straight lines (typically inputs from touch devices) the algorithm simply removes everything but the first and last points. Depending on the speed of drawing on an inertial device there can be additional mid-points preserved (quite rare). This results in either an invalid polygon or less frequently in a truncated polygon like in your case.

**Comment by GaborFarkas** [comment]
I just cloned the repo to create a fix, however, it has been already patched. AFAICS this issue is no longer relevant.

**Comment by trygveaa** [comment]
Yes, I can confirm that this is fixed (tested with 6.3.0 now). Thanks
